build/Common.prod.props

```diff
@@ -32,11 +35,4 @@
     <SymbolPackageFormat>snupkg</SymbolPackageFormat>
   </PropertyGroup>
 
-  <ItemGroup Condition="'$(Deterministic)'=='true'">
-    <SourceRoot Include="$(MSBuildThisFileDirectory)/" />
-  </ItemGroup>
-
-  <PropertyGroup Condition="'$(Deterministic)'=='true'">
-    <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
-  </PropertyGroup>
```

Comment:
Can you explain why you removed this?

Reply:
lol sorry, I see https://github.com/open-feature/dotnet-sdk/pull/173#issuecomment-1901244232 now

Reply:
I removed `L39-L41` because I realized that my changes at the top of this file were already setting it. Those changes condition both `<ContinuousIntegrationBuild>` and `<Deterministic>` on the [`CI` env var which GitHub Actions always sets to `true`](https://docs.github.com/actions/learn-github-actions/variables#default-environment-variables) ensuring these properties are set by default for _all_ `dotnet` commands we run anywhere in our CI workflows.

> [!Note]
>
> _A happy side-effect of conditioning these on the `CI` env var is that it makes testing this stuff locally much easier, because you can just prefix your commands, e.g. `CI=true dotnet pack`, and everything triggers accordingly._
>
> _This is admittedly not dramatically different from running `dotnet pack -p:Deterministic=true`, but IMO its easier to grok these things as "when in CI, do ..." vs. "when deterministic do ..."._
>
> _This^ might not be a terribly compelling reason on its own, but in conjunction with knowing that every `dotnet` command we run in any of our workflows will always have these flags set correctly, without having to remember to pass `-p:Deterministic=true`, is, I think, nice to have._

```diff
diff --git a/build/Common.prod.props b/build/Common.prod.props
index 63a52d4..4d073ec 100644
--- a/build/Common.prod.props
+++ b/build/Common.prod.props
@@ -2,7 +2,10 @@
   <Import Project=".\Common.props" />

   <PropertyGroup>
+    <ContinuousIntegrationBuild Condition="'$(CI)' == 'true'">true</ContinuousIntegrationBuild>
+    <Deterministic Condition="'$(CI)' == 'true'">true</Deterministic>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <PackRelease>true</PackRelease>
   </PropertyGroup>

[...]
```

___

As for why I had to remove `<SourceRoot>`, that I'm less sure about. When testing the packing step locally (i.e. `CI=true dotnet pack ...`), the resultant package was being reported as _not_ having Source Link enabled correctly, according to [NuGetPackageExplorer](https://github.com/NuGetPackageExplorer/NuGetPackageExplorer).

Removing that block and then re-running `CI=true dotnet pack ...` produced a package that NuGetPackageExplorer reported as having Source Link enabled.

_(I wish I had a more rigorous justification for this one, but that's all I've got for now...)_

Reply:
> lol sorry, I see [#173 (comment)](https://github.com/open-feature/dotnet-sdk/pull/173#issuecomment-1901244232) now

Ope, silly GitHub didn't show me this comment until after I replied to your question.

All the same, let me know if you want me to get you a better answer on the `SourceLink` one.